coldfront/core/utils/management/commands/audit_data.py

```diff
@@ -0,0 +1,313 @@
+          particular UC Berkeley semester.
+            - Inactive ones should have a start date, but no end date.
+            - Active ones (currently none on production) should match a current
+                (has started, but has not ended) instructional AllocationPeriod.
+          Recharge and Condo may/should have start dates, but don’t have end
+          dates, as they don’t end.
+        '''
+        allocations = Allocation.objects.select_related('project',
+                            'project__status').prefetch_related('resources') \
+            .filter(resources__name__endswith='Compute') \
+            .order_by('resources__name', 'project__name',
+                      'project__status__name', 'start_date', 'end_date') \
+            .values('id', 'project__status__name',
+                   'project__name','start_date', 'end_date',
+                   'resources__name')
+
+        FCA_PCA_ALLOCATION_PERIOD = AllocationPeriod.objects.get(
+                                        name__startswith='Allowance Year',
+                                        start_date__lt=datetime.date.today(),
+                                        end_date__gt=datetime.date.today())
+        ICA_ALLOCATION_PERIODS = AllocationPeriod.objects.filter(
+                                    Q(end_date__gt=datetime.date.today())
+                                    & (Q(name__startswith='Fall Semester')
+                                       | Q(name__startswith='Spring Semester')
+                                       | Q(name__startswith='Summer Sessions') \
+                                      )).all()
+
+        FCA_PCA_PREFIXES = ('fc_', 'pc_')
+        ICA_PREFIX = 'ic_'
+        RECHARGE_CONDO_PREFIX = ('co_', 'ac_')
+
+        for allocation in allocations:
+            id = allocation['id']
+            start_date = allocation['start_date']
+            end_date = allocation['end_date']
+            project_status = allocation['project__status__name']
+            project = allocation['project__name']
+            resource = allocation['resources__name']
+
+            if end_date is not None and end_date < start_date:
+                self.stdout.write(self.style.ERROR('Allocation {} '
+                'for {} project {} has an end date before its start date.' \
+                    .format(allocation['id'], project_status.lower(), project)))
+
+            if project.startswith(FCA_PCA_PREFIXES):
+                if project_status == 'Inactive' \
+                        and start_date != FCA_PCA_ALLOCATION_PERIOD.start_date:
+                    self.stdout.write(self.style.ERROR('{} {} '
+                    'for inactive FCA or PCA project {} has a start date of {} '
+                    'that is different than its allocation period\'s ({}).' \
+                        .format(resource, id, project,
+                            start_date, FCA_PCA_ALLOCATION_PERIOD.start_date)))
+                if project_status == 'Active' \
```

Comment:
We could also assert that the start date is no earlier than the start date of the allocation period.